lib/src/androidTest/java/com/couchbase/lite/internal/core/C4MutableFleeceTest.java

```diff
@@ -50,27 +50,38 @@
 
 public class C4MutableFleeceTest extends C4BaseTest {
 
+    private MValue.Delegate delegate;
+
     @Before
     public void setUp() throws Exception {
         super.setUp();
+        delegate = MValue.getRegisteredDelegate();
         MValue.registerDelegate(new MValueDelegate());
     }
 
     @After
     public void tearDown() throws Exception {
+        if (delegate != null)
+            MValue.registerDelegate(delegate);
         super.tearDown();
     }
 
     static AllocSlice encode(Object obj) throws LiteCoreException {
-        Encoder enc = new Encoder();
-        enc.writeObject(obj);
-        return enc.finish();
+        FLEncoder enc = new FLEncoder();
+        try {
+            enc.writeValue(obj);
+            return enc.finish2();
```

Comment:
what do you think about adding a comment saying something like, result should be freed after use.